modules/tensor_mechanics/src/kernels/TotalLagrangianStressDivergence.C

```diff
@@ -0,0 +1,209 @@
+#include "TotalLagrangianStressDivergence.h"
```

Comment:
This kernel should be separate into two kernels. One responsible for the total lagrangian stress divergence, the other one for the homogenization constraint. The homogenization constraint kernel can either be a separate kernel or inherit from `TotalLagrangianStressDivergence`. Inheritance probably makes more sense because it can only be used in conjunction with this total lagrangian formulation.

Reply:
Agreed!  Both the kernel and strain calculator classes get split up now.